experimental/AlgebraicStatistics/src/DiscreteGraphicalModels.jl

```diff
@@ -190,109 +189,136 @@ end
+  to multivariate polynomial ring in 13 variables over QQ
 defined by
-p[1, 1, 1] -> t[1, 2](1, 1)*t[2, 3](1, 1)
-p[2, 1, 1] -> t[1, 2](2, 1)*t[2, 3](1, 1)
-p[1, 2, 1] -> t[1, 2](1, 2)*t[2, 3](2, 1)
-p[2, 2, 1] -> t[1, 2](2, 2)*t[2, 3](2, 1)
-p[1, 1, 2] -> t[1, 2](1, 1)*t[2, 3](1, 2)
-p[2, 1, 2] -> t[1, 2](2, 1)*t[2, 3](1, 2)
-p[1, 2, 2] -> t[1, 2](1, 2)*t[2, 3](2, 2)
-p[2, 2, 2] -> t[1, 2](2, 2)*t[2, 3](2, 2)
+  p[1,1,1] -> q[1](1)*q[2](1)*_h - q[1](1)*q[3](2 | 1, 1)*_h - q[1](2)*q[2](2)*q[3](2 | 1, 1) + q[2](2)*q[3](2 | 1, 1)*_h
+  p[2,1,1] -> q[1](2)*q[2](1)*_h + q[1](2)*q[2](2)*q[3](2 | 1, 2) - q[1](2)*q[3](2 | 1, 2)*_h
+  p[1,2,1] -> q[1](1)*q[2](2)*_h + q[1](2)*q[2](2)*q[3](2 | 2, 1) - q[2](2)*q[3](2 | 2, 1)*_h
+  p[2,2,1] -> -q[1](2)*q[2](2)*q[3](2 | 2, 2) + q[1](2)*q[2](2)*_h
+  p[1,1,2] -> q[1](1)*q[3](2 | 1, 1)*_h + q[1](2)*q[2](2)*q[3](2 | 1, 1) - q[2](2)*q[3](2 | 1, 1)*_h
+  p[2,1,2] -> -q[1](2)*q[2](2)*q[3](2 | 1, 2) + q[1](2)*q[3](2 | 1, 2)*_h
+  p[1,2,2] -> -q[1](2)*q[2](2)*q[3](2 | 2, 1) + q[2](2)*q[3](2 | 2, 1)*_h
+  p[2,2,2] -> q[1](2)*q[2](2)*q[3](2 | 2, 2)
+
+julia> I = kernel(phi)
+Ideal generated by
+  -p[1,1,1]*p[2,2,1] - p[1,1,1]*p[2,2,2] + p[2,1,1]*p[1,2,1] + p[2,1,1]*p[1,2,2] + p[1,2,1]*p[2,1,2] - p[2,2,1]*p[1,1,2] - p[1,1,2]*p[2,2,2] + p[2,1,2]*p[1,2,2]
 """
-# function parametrization(M::GraphicalModel{Graph{Directed}, MarkovRing})
-# 
-#   G = graph(M)
-#   V = vertices(G)
-#   S = ring(M)
-#   rvs = random_variables(S)
-#   R = param_ring(M)
-#   q = param_gens(M)
-# 
-#   images = []
-# 
-#   for ind in state_space(S)
-# 
-#       push!(images, prod(map(i-> q[i, ind[i]][Tuple(ind[inneighbors(G, i)])], V)))
-#   end
-# 
-#   lin_constraints  = []
-# 
-#   for i in V
-#     for par_state in parental_state_space(G, i, S)
-# 
-#       push!(lin_constraints, last(gens(R)) - sum(map(j -> q[i, j][Tuple(par_state)] , state_space(S, rvs[i]))))
-#     end
-#   end
-#   I = ideal(lin_constraints)
-# 
-#   return hom(ring(S), R, map(f -> reduce(f, gens(I)), images))
-# end
+function parametrization(M::DiscreteGraphicalModel{Graph{Directed}, T}) where T
+  G = graph(M)
+  n = n_vertices(G)
+  S, pd = model_ring(M)
+  R, qd = parameter_ring(M)
+  h = last(gens(R))
+  images = []
+  for p in gens(_ring(S))
+    s = [S[p]...] # get label of the variable but as a vector for type reasons
+    push!(images, prod(qd[(i, s[i], s[parents(G, i)])] for i in 1:n))
+  end
+  # There are linear constraints to enforce on the `images`. Instead of
+  # doing this by hand, we employ a Gröbner basis of the linear ideal.
+  condprob = []
+  for i in 1:n
+    pa = parents(G, i)
+    if length(pa) == 0
+      push!(condprob, h - sum(qd[(i, x, Int64[])] for x in state_space(M, [i])))
+    else
+      for y in state_space(M, pa)
+        push!(condprob, h - sum(qd[(i, x, collect(Iterators.flatten([y])))] for x in state_space(M, [i])))
+      end
+    end
+  end
+  I = ideal(condprob)
+  return hom(S, R, map(f -> reduce(f, gens(I)), images))
```

Comment:
might not be a big deal here so feel free to ignore, but in general it's best to avoid anonymous functions.
```suggestion
  return hom(S, R, map(Base.fix2(reduce, gens(I)), images))
```